config/helm/chart/default/templates/_helpers.tpl

```diff
@@ -118,45 +110,21 @@ Check if default image is used
 {{- end -}}
 {{- end -}}
 
-{{/*
-Check if only 1 oneagent mode is used.
-*/}}
-{{- define "dynatrace-operator.modeSet" -}}
-	{{- $modes := list .Values.cloudNativeFullStack .Values.classicFullStack .Values.hostMonitoring .Values.applicationMonitoring -}}
-	{{- $enabled := dict -}}
-		{{- range $index, $mode := $modes -}}
-			{{- if $mode -}}
-			{{- if $mode.enabled -}}
-				{{- $_ := set $enabled ($index|toString) ($mode|toString) -}}
-			{{- end -}}
-			{{- end -}}
-		{{- end -}}
-		{{- if (lt (len (keys $enabled)) 2 ) -}}
-			{{- "set" -}}
-		{{- end -}}
-{{- end -}}
-
 
 {{/*
 Check if we need the csi driver.
 */}}
 {{- define "dynatrace-operator.needCSI" -}}
-	{{- if eq (include "dynatrace-operator.partial" .) "csi" -}}
+	{{- if eq (default false .Values.csidriver.enabled) true -}}
```

Comment:
```suggestion
	{{- if or (.Values.csidriver.enabled) (eq (include "dynatrace-operator.partial" .) "csi") -}}
```

Reply:
I'd prefer if the if conditions are merged as they return the same thing.
I guess the default could be removed since the zero value of a bool is false in any case